### PR TITLE
Vehicles

### PR DESCRIPTION
**Pull request by Addison118**
Added Vehicle events

### CODE REVIEW

File: src/pocketmine/event/vehicle/VehicleDamageEvent.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	public function getAttacker() {
+		return $this->attacker;
+	}
+	
+	public function getDamage() {
+		return $this->damage;
+	}
+	
+	public function setDamage($damage) {
+		setDamage($damage);
```
Comment (iksaku) [inline src/pocketmine/event/vehicle/VehicleDamageEvent.php]: I think you forgot a ``$`` at the beginning of this line :3

**Comment by iksaku** [comment]
May I ask what's the purpose of all this events without real ``Vehicles`` implementation?

**Comment by Addison118** [comment]
@iksaku I thought that I should add the events before I add the actual vehicles.

**Comment by Addison118** [comment]
It also makes it easier for anyone else that wants to add the minecarts or boats. I just made it easier by adding the events. I will start working on the actual vehicles tomorrow.

**Comment by iksaku** [comment]
Well then, just make sure to properly code everything ;) you have some basic PHP mistakes on your code :S Please use and IDE and always check for possible errors :P

**Comment by Addison118** [comment]
Yeah I probably do. What's the one you were talking about up there? I forgot a $?

**Comment by Addison118** [comment]
Ah. $this->damage = $damage

**Comment by Addison118** [comment]
;

**Comment by Gamecrafter** [comment]
Not again @Addison118.

**Comment by Addison118** [comment]
@GameCrafter do you have a problem? If you have nothing helpful to say then don't comment :)

**Comment by iksaku** [comment]
@Gamecrafter Then don't try to make plugins please... You don't know a lot too :S 

@Addison118 Please read the PocketMine API carefully... You're overriding functions like ``isCancelled()``and ``setCancelled()`` on events, they're already set, to implement them just write ``implements Cancellable`` in the class declaration line

**Comment by Addison118** [comment]
@iksaku but there is nothing wrong with overriding them correct?

**Comment by iksaku** [comment]
You shouldn't override them, because the other ones has some protection code, and yours not :$

**Comment by Addison118** [comment]
@iksaku ok I will close this and fix all the small errors :3

**Comment by sekjun9878** [comment]
1. What's the point of inheritance if you're just going to redeclare methods in child classes?
2. Looks like you need an IDE to do some static code analysis because there are some code errors.
3. Squash your commits! We don't want 30 commits for each file creation messing up our commit history!

And I think shoghicp probably has his own implementation in mind and boilerplate codes like this one don't really add any valuable features to PM. But thanks for contributing to the project!

**Comment by Addison118** [comment]
@sekjun9878 XD i'm using netbeans right now (Seems like the linux version doesn't have all the features that windows has :p) and I will squash my commits next time... Seems like there are other important things to do so I thought someone should do it.

**Comment by PEMapModder** [comment]
@Addison118 it appears that your knowledge on object-oriented programming as in PHP 5 is very inadequate. Please note that PocketMine-MP, since Zekkou Cake, from internal framework to external API, is very object-oriented. It seems like you don't even understand inheritance.

Also, most of the events that you created are already covered by existing events such as `EntityDamageEvent`.

About only declaring events, it appears that you are only creating events whenever they come to your mind. Without the real code being written, creating the events before writing the real code only makes coding more difficult, because events are usually closely related to how the code is written. For example, you won't realize that you do not want to let plugins cancel some specific events until you are actually making it. Furthermore, PocketMine API is in some ways based on/modelling the Bukkit API.

On the other hand, please carefully review the [contribution guidelines](../blob/master/CONTRIBUTING.md) before creating a pull request.

@iksaku @Addison118 @Gamecrafter please note that GitHub is not a forum. If you are going on flame wars, find a real battlefield. GitHub is too cool to withstand flame.

**Comment by Addison118** [comment]
@PEMapModder Are you suggesting I should learn the whole PocketMine api? I know object oriented php. I didn't re create the vehicle event in all of my other events. :p

**Comment by PEMapModder** [comment]
1. If you know well about object oriented PHP, you wouldn't override your own inherited functions.
2. VehicleDamageEvent is a duplicate of EntityDamageEvent.

**Comment by Addison118** [comment]
@PEMapModder @iksaku @sekjun9878 You know what guys I was just trying to help pocketmine now see what's happening? Maybe I shouldn't contribute anymore and I'm tired of this happening. I was just trying to get vehicles working because it seems like no one else is making a move. I'm not the best (far from it as you can tell) now can you suggest anything to me whether it be a book or anything like that?

**Pull request closed without merging**